Source/JavaScriptCore/inspector/InspectorAgentBase.h

```diff
@@ -84,4 +84,25 @@ class InspectorAgentBase {
     String m_name;
 };
 
+class InspectorEnvironmentClient {
+protected:
+    explicit InspectorEnvironmentClient(AgentContext& context)
+        : m_environment { context.environment.ptr() }
+    {
+    }
+
+    explicit InspectorEnvironmentClient(InspectorEnvironment* environment)
+        : m_environment { environment }
+    {
+    }
+
+    InspectorEnvironment* environment() { return m_environment.get(); }
+    const InspectorEnvironment* environment() const { return m_environment.get(); }
+    CheckedRef<InspectorEnvironment> checkedEnvironment() { return *environment(); }
+    CheckedRef<const InspectorEnvironment> checkedEnvironment() const { return *environment(); }
```

Comment:
it seems very odd that we convert from a `WeakRef` to a `CheckedRef`.  is that normal?  i thought the idea was that we'd always only use one or the other